crypto/cmp/cmp_genm.c

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright 2022 The OpenSSL Project Authors. All Rights Reserved.
+ * Copyright Siemens AG 2022
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include "cmp_local.h"
+#include <openssl/cmp_util.h>
+
+static const X509_VERIFY_PARAM *get0_trustedStore_vpm(const OSSL_CMP_CTX *ctx)
+{
+    const X509_STORE *ts = OSSL_CMP_CTX_get0_trustedStore(ctx);
+
+    return ts == NULL ? NULL : X509_STORE_get0_param(ts);
+}
+
+static void cert_msg(const char *func, const char *file, int lineno,
+                     OSSL_CMP_severity level, OSSL_CMP_CTX *ctx,
+                     const char *source, X509 *cert, const char *msg)
+{
+    char *subj = X509_NAME_oneline(X509_get_subject_name(cert), NULL, 0);
+
+    ossl_cmp_print_log(level, ctx, func, file, lineno,
+                       level == OSSL_CMP_LOG_WARNING ? "WARN" : "ERR",
+                       "certificate from '%s' with subject '%s' %s",
+                       source, subj, msg);
+    OPENSSL_free(subj);
+}
+
+static int ossl_X509_check(OSSL_CMP_CTX *ctx, const char *source, X509 *cert,
+                           int type_CA, const X509_VERIFY_PARAM *vpm)
+{
+    uint32_t ex_flags = X509_get_extension_flags(cert);
+    int res = X509_cmp_timeframe(vpm, X509_get0_notBefore(cert),
+                                 X509_get0_notAfter(cert));
+    int ret = res == 0;
+    OSSL_CMP_severity level =
+        vpm == NULL ? OSSL_CMP_LOG_WARNING : OSSL_CMP_LOG_ERR;
+
+    if (!ret)
+        cert_msg(OPENSSL_FUNC, OPENSSL_FILE, OPENSSL_LINE, level, ctx,
+                 source, cert, res > 0 ? "has expired" : "not yet valid");
+    if (type_CA >= 0 && (ex_flags & EXFLAG_V1) == 0) {
+        int is_CA = (ex_flags & EXFLAG_CA) != 0;
+
+        if ((type_CA == 1) != is_CA) {
+            cert_msg(OPENSSL_FUNC, OPENSSL_FILE, OPENSSL_LINE, level, ctx,
+                     source, cert,
+                     is_CA ? "is not an EE cert" : "is not a CA cert");
+            ret = 0;
+        }
+    }
+    return ret;
+}
+
+static int ossl_X509_check_all(OSSL_CMP_CTX *ctx, const char *source,
+                               STACK_OF(X509) *certs,
+                               int type_CA, const X509_VERIFY_PARAM *vpm)
+{
+    int i;
+    int ret = 1;
+
+    for (i = 0; i < sk_X509_num(certs /* may be NULL */); i++)
+        ret = ossl_X509_check(ctx, source,
+                              sk_X509_value(certs, i), type_CA, vpm)
+            && ret; /* Having 'ret' after the '&&', all certs are checked. */
```

Comment:
Nit: IMO && should be aligned with ossl_...

Reply:
Thanks for your comments!

`check-format.pl crypto/cmp/cmp_genm.c`, as well as the Emacs C mode, is happy with this indentation.
The idea is that on a multi-line assignment (and in similar other cases),
indentation is increased by the usual delta (+4), which is independent of the length of the variable name on the LHS.

Reply:
I'd have indented this as @t8m suggests

Reply:
I've reworked the assignment to generally make this code more readable.
As a by-product, the indentation issue disappears.